### DISCUSSION TITLE
Trying to lock post blocks on load and restore lock state on save

### DISCUSSION THREAD

**Discussion by grgsllnd** [WordPress/gutenberg#49481] [Q&A]
I found the way to Lock blocks programmatically:
```
const lockBlocks = createHigherOrderComponent( ( Block ) => {
    return ( props ) => {
        if(!(props.attributes.lock && props.attributes.lock.move==true && props.attributes.lock.remove==true)){
            console.warn('locking')
            props.attributes.old_lock=props.attributes?.lock||null
            //props.attributes.templateLock=true //is there something like that to do?
            props.attributes.lock={
                        move:true,
                        remove:true,
                         //isLocked:true, //maybe instead of "old_lock"
            }
		
        }
    }
    return el(Block,{...props})
    };
}, 'lockBlocks');
addFilter('editor.BlockListBlock','URBEL/lock-blocks',lockBlocks);
```
But when blocks are saved, they keep the lock state
I can't find a filter or any way to restore the block lock sate on save
I would like to restore the state with props.attributes.old_lock
Thanks in advance
I already tried 'blocks.getSaveContent.extraProps' and 'blocks.getSaveElement' filters without success
Is there a filter on blocks content/attributes during save?
(Something like https://developer.wordpress.org/reference/hooks/wp_insert_post_data/ in php but for gutenberg in js)
Maybe my approach to the problem is wrong

I also tried 

```
//alternative php way
function filter_block_editor_settings_when_post_provided( $editor_settings, $editor_context ) {
        //echo '<pre>';var_dump($editor_settings);exit;
        return $editor_settings;
        }//add_filter( 'block_editor_settings_all', 'filter_block_editor_settings_when_post_provided', 10, 2 );

        function filter_metadata_registration( $settings, $metadata ) {
           //echo '<pre>';var_dump(['settings'=>$settings, 'metadata'=>$metadata]);exit;
           if(!isset( $settings["attributes"])) $settings["attributes"]=[];
           $settings["attributes"]["lock"]=[
           "type"=>"object",
           "default"=>[
           "remove"=>true,
           "move"=>true
           ]
           ];
           return $settings;
       }
add_filter( 'block_type_metadata_settings', 'filter_metadata_registration', 10, 2 );
```
and 

```
function myplugin_register_template() {
    $post_type_object = get_post_type_object( 'post' );
    $post_type_object->template =[...];
    $post_type_object->template_lock = 'all';
}
add_action( 'init', 'myplugin_register_template' );
```
add_action( 'init', 'myplugin_register_template' );

but this seems to only lock new created posts with the template defined

**Comment by Mamaduka** [comment]
Thank you, @GregoireSailland, for opening a separate discussion.

Can you share more details about the use case and what you plan to achieve? Maybe there's a different way to do it.

**Reply by grgsllnd** [reply]
The idea is to block layout for some users by role or capability.
For exemple you want website editors to write content and set layout for pages and you want translators for exemple with a plugin like polylang to can translate posts contents without allowing them to change the layout or delete some blocks by mistake
Still there is a caveat in this process: Some languages don't have the same order in languages and some links would be cuting sentences in wrong place. But generally it works fine.
I already developped a custom plugin for translating strings found in gutenberg default language post and replacing them in translated post content from default language, but in a custom admin vuejs app.
I would like to do the same now with gutenberg now.

An other use case of this is if you want to prevent some users to modify layout but allow them to modify pages text content or images.

The block lock feature is perfect for that but not conditional by user. You maybe don't want to block layout for everybody.
Maybe you want your graphists to create pages layouts with lorem ipsum but allow editors to change content only.
